controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -702,6 +704,55 @@ private void handleStreamMetadataExists(final long timestamp, CompletableFuture<
         return result;
     }
 
+    @Override
+    CompletableFuture<Void> createSealedSegmentsRecord(byte[] sealedSegmentRecord) {
+        Preconditions.checkNotNull(sealedSegmentRecord);
+
+        CompletableFuture<Void> result = new CompletableFuture<>();
+
+        synchronized (lock) {
+            this.sealedSegments = new Data<>(sealedSegmentRecord, 0);
+            result.complete(null);
+        }
+        return result;
+    }
+
+    @Override
+    CompletableFuture<Data<Integer>> getSealedSegmentsRecord() {
+        CompletableFuture<Data<Integer>> result = new CompletableFuture<>();
```

Comment:
Similarly to above, it's perfectly fine to return `Futures.failedFuture` or `CompletableFuture.completedFuture` with the result directly from the `synchronized` block.

Reply:
done